### ISSUE TITLE
HTML inside code blocks is rendered

### ISSUE THREAD

**Issue by systimotic** [freeCodeCamp/guide#587]
If there is HTML (or something that resembles an HTML tag) inside of a Markdown code block, it is not escaped. It is only escaped if the language of that code block is set to HTML, or a different language that is set up to handle things that look like HTML elements.

This causes issues with HTML inside strings inside Javascript code blocks, and primarily with all of the code blocks that contain HTML but are not marked as such (there are a lot of them).

This issue is related to some problems with `gatsby-remark-prismjs`, which we use for syntax highlighting. This error doesn't happen without it.

Example (ignore all `~`, they are there to make GitHub show the things I want it to show):
```
```html
<div>
  This is not rendered as html, but as code. This is what we want.
</div>
``` ~
```

```
``` ~
<div>
  This is rendered as html, not as code. This usually messes up the page.
</div>
``` ~
```

Related: #233 #239 #246 #275 #450 #453

Still an issue with a lot of articles:
- https://guide.freecodecamp.org/certificates/create-a-bootstrap-headline
- https://guide.freecodecamp.org/certificates/create-a-bootstrap-row
- https://guide.freecodecamp.org/certificates/target-html-elements-with-selectors-using-jquery
- https://guide.freecodecamp.org/certificates/use-css-selectors-to-style-elements
- https://guide.freecodecamp.org/certificates/use-hex-code-to-color-elements-blue (:smile:)

And there are a lot more around there.

I think we should probably open an issue at Gatsby about this. I haven't been able to find anything about it over there.
/cc @Bouncey

**Comment by Ethan-Arrowood** [comment]
Have we considered a stricter contributing guideline regarding embedded code snippets? I always put the language after the ` ``` ` when I write markdown; we should try to make it a forced practice. 

This would require us manually fixing all currently broken embeds, and maybe we could add something to the precommit file to check for a language (or at least some string) after ` ``` ` in any .md file.

**Comment by systimotic** [comment]
@Ethan-Arrowood I think it would be a great idea to more strongly advice people to specify the language, but looking at what we're currently saying (https://github.com/freeCodeCamp/guides#formatting-example-code), I'm not sure how to improve it.

Most people create their articles on GitHub and never therefore never run the pre-commit hook. It would be nice to find some way to enforce this, but manually is probably the best way for now.

The majority of articles in the Guide were imported from the forum. This issue didn't exist on the forum, so a lot of articles were created using the indented syntax to create a code block, or didn't specify a language. It would be great to go over all of these, but even in the `Certificates` folder alone there are 260 articles. Going over all of them and assigning a language would take a lot of time.

All of this would be an amazing improvement, but we'd continue to see the underlying issue in cases like #239, where something that has the same syntax as an HTML element is genuinely part of a different language. Example:
```javascript
console.log('This is totally a JavaScript code block');
console.log(['it.', 'about', 'question', 'No'].reverse().join(' '));
document.body.insertAdjacentHTML('beforeend', '<div>But this is also valid JavaScript.<span>This HTML would be rendered, but it shouldn\'t be.</span></div>')

**Comment by Ethan-Arrowood** [comment]
Hmm this is a tricky one then. I imagine we could find a way to improve Gatsby to suit our needs. I think you should open an issue on Gatsby at least.

**Comment by nsuchy** [comment]
Does this happen even when you use,
```
```html
```
to start your code block? For now we can at least limit the issues impact as maintainers, however I'd be interested in finding which code makes html not render when using the above method and backporting to the plain text code blocks.

**Comment by smsrkr** [comment]
I can confirm adding a syntax name after the starting ` ``` ` ~~solves~~ works around this issue.

**Comment by erictleung** [comment]
## **Note**

To all planning on working on this, please make small, manageable changes so it'll be easier for you to make changes and for us to review it.

In your pull request, **do not** have `Fix #587` in your pull request unless it really is the last set of fixes. Please **do** have something along the lines of `Ref #587` (so that it doesn't automatically close this issue) in your pull request so that we can track the progress of this issue. See [here](https://help.github.com/articles/closing-issues-using-keywords/) for a list of words not to use.

Feel free to check out https://github.com/freeCodeCamp/guides/pull/8086 for the kind of changes we're looking for. We are grateful to @smsrkr for putting in a lot of effort into pointing out practically all the files that need changes.

Let us know if you have any questions. Happy coding!
